torch_geometric/nn/pool/gm_pool.py

```diff
@@ -0,0 +1,291 @@
+    return idx, inverse
+
+
+@torch.no_grad()
+def build_knn_graph(src_emb: Tensor, dst_emb: Tensor, src_batch: Tensor,
+                    dst_batch: Tensor, k: int) -> Tensor:
+    src_idx, src_inverse = argsort(src_batch)
+    dst_idx, dst_inverse = argsort(dst_batch)
+    graph = knn(dst_emb[dst_idx], src_emb[src_idx], k, dst_batch[dst_idx],
+                src_batch[src_idx])
+    return torch.stack([src_inverse[graph[0]], dst_inverse[graph[1]]], dim=0)
+
+
+class GMPooling(torch.nn.Module):
+    r"""The Gaussion Mixture Pooling implementation from
+    `"PHierarchical Graph Neural Networks for Particle Track Reconstruction"
+    <https://arxiv.org/abs/2303.01640>`_ paper.
+
+    In short, GMPooling pools the graph by computing the edge scores defined
+    as :math:`\tanh^{-1}(x_i\cdot x_j)`, where :math:`x_i` are normalized node
+    embeddings. Assuming that there are intra-cluster and inter-cluster edges,
+    we fit a Gaussian Mixture Model of 2 components.The score cut is solved by
+    finding the score where the difference in log likelihood of the two
+    components exceeds a hyperparameter :obj:`r` that is used to tune the
+    granularity of the pooling. Finally, the connected components that have
+    more than :obj:`min_size` are regarded as the pooled nodes, which are
+    called "supernodes" in the paper.
+
+    GMPooling takes in node embeddings, graph edges, and batch indices and
+    returns pooled embeddings (supernodes) and their batch indices.
+    Optionally, GMPooling can also return bipartite graph in between the
+    original nodes and the newly created supernodes, together with edge
+    weights that ensure the differentiability of the method, and the super
+    graph on the supernodes and their edge weights.
+
+    Args:
+        r (int): Resolution which controls the granularity of the pooling.
+        min_size (int): Minimum size for a connected component to be kept as
+            supernode. Any component smaller than this will be treated as
+            noise.
+        build_bipartite_graph (bool, optional): Whether to build bipartite
+            graph or not. When set to :obj:`True`, edges and edge weights are
+            returned. (default: :obj:`False`)
+        build_super_graph (bool, optional): Whether to build super graph or
+            not. When set to :obj:`True`, edges and edge weights are returned.
+            (default: :obj:`False`)
+        bipartite_k (int, optional): The connectivity of the bipartite graph
+            being built. (default: :obj:`5`)
+        super_k (int, optional): The connectivity of the super graph being
+            built. (default: :obj:`5`)
+        momentum (float, optional): The momentum of the exponential moving
+            average used to track score cut. (default: :obj:`0.95`)
+    """
+    def __init__(self, r: float, min_size: int,
+                 build_bipartite_graph: Optional[bool] = False,
```

Comment:
```suggestion
                 build_bipartite_graph: bool = False,
```
Not need to be `optional`.